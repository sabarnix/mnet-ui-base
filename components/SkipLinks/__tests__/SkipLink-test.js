"use strict";

var _react = _interopRequireDefault(require("react"));

require("jest-styled-components");

var _react2 = require("@testing-library/react");

var _ = require("../..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

describe('SkipLink', function () {
  afterEach(_react2.cleanup);
  test('basic', function () {
    jest.useFakeTimers();

    var _render = (0, _react2.render)( /*#__PURE__*/_react["default"].createElement(_.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.SkipLinks, {
      id: "skip-links"
    }, /*#__PURE__*/_react["default"].createElement(_.SkipLink, {
      id: "main",
      label: "Main Content"
    }), /*#__PURE__*/_react["default"].createElement(_.SkipLink, {
      id: "footer",
      label: "Footer"
    })), /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_.SkipLinkTarget, {
      id: "main"
    }), "Main Content", /*#__PURE__*/_react["default"].createElement("input", {
      type: "text",
      value: "main content",
      onChange: function onChange() {}
    })), /*#__PURE__*/_react["default"].createElement("footer", null, /*#__PURE__*/_react["default"].createElement(_.SkipLinkTarget, {
      id: "footer"
    }), /*#__PURE__*/_react["default"].createElement("input", {
      type: "text",
      value: "footer",
      onChange: function onChange() {}
    })))),
        container = _render.container;

    expect(container.firstChild).toMatchSnapshot();
    document.getElementById('skip-links').querySelector('a').focus();
    expect(container.firstChild).toMatchSnapshot();

    _react2.fireEvent.click(document.activeElement);

    document.getElementById('skip-links').querySelector('a').blur();
    (0, _react2.act)(function () {
      jest.runAllTimers();
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});