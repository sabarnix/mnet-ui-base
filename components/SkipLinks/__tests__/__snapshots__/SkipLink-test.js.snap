// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink basic 1`] = `
.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #E15151;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  width: 0;
  height: 0;
  overflow: hidden;
  position: absolute;
}

@media only screen and (max-width:512px) {

}

@media only screen and (max-width:512px) {

}

@media only screen and (max-width:512px) {

}

@media only screen and (max-width:512px) {

}

@media only screen and (max-width:512px) {

}

@media only screen and (max-width:512px) {

}

<div
  class="c0"
>
  <div>
    <a
      aria-hidden="true"
      class="c1 c2"
      id="main"
      tabindex="-1"
    />
    Main Content
    <input
      type="text"
      value="main content"
    />
  </div>
  <footer>
    <a
      aria-hidden="true"
      class="c1 c2"
      id="footer"
      tabindex="-1"
    />
    <input
      type="text"
      value="footer"
    />
  </footer>
</div>
`;

exports[`SkipLink basic 2`] = `
<div
  class="StyledMnetUIBase-sc-17lh5y2-0 HlCER"
>
  <div>
    <a
      aria-hidden="true"
      class="StyledAnchor-sc-1meb4kf-0 xsVdC SkipLinkTarget__HiddenAnchor-sc-1cistln-0 bZSZHG"
      id="main"
      tabindex="-1"
    />
    Main Content
    <input
      type="text"
      value="main content"
    />
  </div>
  <footer>
    <a
      aria-hidden="true"
      class="StyledAnchor-sc-1meb4kf-0 xsVdC SkipLinkTarget__HiddenAnchor-sc-1cistln-0 bZSZHG"
      id="footer"
      tabindex="-1"
    />
    <input
      type="text"
      value="footer"
    />
  </footer>
</div>
`;

exports[`SkipLink basic 3`] = `
<div
  class="StyledMnetUIBase-sc-17lh5y2-0 HlCER"
>
  <div>
    <a
      aria-hidden="true"
      class="StyledAnchor-sc-1meb4kf-0 xsVdC SkipLinkTarget__HiddenAnchor-sc-1cistln-0 bZSZHG"
      data-g-tabindex="-1"
      id="main"
      tabindex="-1"
    />
    Main Content
    <input
      data-g-tabindex="none"
      tabindex="-1"
      type="text"
      value="main content"
    />
  </div>
  <footer>
    <a
      aria-hidden="true"
      class="StyledAnchor-sc-1meb4kf-0 xsVdC SkipLinkTarget__HiddenAnchor-sc-1cistln-0 bZSZHG"
      data-g-tabindex="-1"
      id="footer"
      tabindex="-1"
    />
    <input
      data-g-tabindex="none"
      tabindex="-1"
      type="text"
      value="footer"
    />
  </footer>
</div>
`;
